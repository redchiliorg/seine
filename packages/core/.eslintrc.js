module.exports = {
  extends: [
    'react-app',
    'plugin:import/recommended',
    'plugin:prettier/recommended',
    'plugin:jsdoc/recommended',
  ],
  rules: {
    'import/order': ['error', { 'newlines-between': 'always' }],
    'no-console': 'error',
    'react-hooks/exhaustive-deps': 'warn',
    'react-hooks/rules-of-hooks': 'error',
    'jsdoc/require-returns-description': 0,
    'jsdoc/require-param-description': 0,
    'jsdoc/require-description': ['warn', { descriptionStyle: 'tag' }],
  },
};
