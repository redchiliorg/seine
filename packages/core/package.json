{
  "name": "@seine/core",
  "version": "0.3.70",
  "description": "Seine core utils and types.",
  "main": "lib/cjs/index.js",
  "module": "lib/esm/index.js",
  "files": [
    "lib/"
  ],
  "repository": "redchiliorg/seine",
  "author": "Mikhail Boyarov",
  "license": "MIT",
  "dependencies": {
    "uuid": "^3.3.2"
  }
}
