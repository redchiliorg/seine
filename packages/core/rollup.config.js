import config from '../../rollup.config';

export default {
  ...config,
  external: [...config.external, 'crypto'],
};
