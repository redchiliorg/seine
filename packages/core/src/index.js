// @flow
import * as mocks from './mocks';

export { mocks };
export * from './reducers';
export * from './types';
export * from './hooks';
