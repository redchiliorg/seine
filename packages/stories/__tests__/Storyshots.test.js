import initStoryshots from '@storybook/addon-storyshots';
import * as core from '@material-ui/core';

jest.mock('@material-ui/core');
core.Popover = jest.fn().mockReturnValue(null);

jest.mock('uuid/v4', () => jest.fn().mockReturnValue(null));

initStoryshots({
  storyKindRegex: /^(?!Admin).+/,
});
