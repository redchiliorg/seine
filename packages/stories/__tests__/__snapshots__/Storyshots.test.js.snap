// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Bream Bar Chart In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Column And Line Charts In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Column Chart In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Column Pie And Line Charts In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Column Pie Bar And Line Charts In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Content In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Line Chart In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Pie Chart In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Pie Chart Legend In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Rich Text In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Table In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Two Pie Charts In BCG Layout 1`] = `null`;

exports[`Storyshots Bream Two Tables Grid In BCG Layout 1`] = `null`;

exports[`Storyshots Bream.Review summary header Bad Performance 1`] = `null`;

exports[`Storyshots Bream.Review summary header Good Performance 1`] = `null`;

exports[`Storyshots Bream.Review summary header Great Performance 1`] = `null`;

exports[`Storyshots Chart constraints Bar Chart On Small Screens 1`] = `null`;

exports[`Storyshots Chart constraints Column Chart On Small Screens 1`] = `null`;

exports[`Storyshots Chart constraints Line Chart On Small Screens 1`] = `null`;

exports[`Storyshots Chart constraints Pie Chart On Small Screens 1`] = `null`;

exports[`Storyshots Default.Multiple.Content Content Of All Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Content Content Of Chart Sample One 1`] = `null`;

exports[`Storyshots Default.Multiple.Content Content Of Column And Bar Sibling Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Content Content Of Column And Line Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Content Content Of Line And Bar Sibling Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Content Content Of Pie And Bar Sibling Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Content Content Of Six Groups Of Two Columns And Bar Sibling Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Content Initial Content 1`] = `null`;

exports[`Storyshots Default.Multiple.Editor Editor Of All Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Editor Editor Of Chart Sample One 1`] = `null`;

exports[`Storyshots Default.Multiple.Editor Editor Of Column And Bar Sibling Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Editor Editor Of Column And Line Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Editor Editor Of Line And Bar Sibling Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Editor Editor Of Pie And Bar Sibling Charts 1`] = `null`;

exports[`Storyshots Default.Multiple.Editor Initial Editor 1`] = `null`;

exports[`Storyshots Default.Single.Chart Bar Chart 1`] = `null`;

exports[`Storyshots Default.Single.Chart Column Chart 1`] = `null`;

exports[`Storyshots Default.Single.Chart Dummy Chart Layout 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Bar Chart 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Column Chart 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Line Chart 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Pie Chart 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Six Groups Of Six Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Six Groups Of Two Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Three Groups Of Four Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Three Groups Of Three Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Two Groups Of Six Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Editor Of Two Groups Of Two Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Line Chart 1`] = `null`;

exports[`Storyshots Default.Single.Chart Pie Chart 1`] = `null`;

exports[`Storyshots Default.Single.Chart Six Groups Of Six Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Six Groups Of Two Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Three Groups Of Four Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Three Groups Of Three Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Two Groups Of Six Columns 1`] = `null`;

exports[`Storyshots Default.Single.Chart Two Groups Of Six Columns Min Value 20 Max Value 100 And Dy 5 1`] = `null`;

exports[`Storyshots Default.Single.Chart Two Groups Of Six Columns Min Value Minus 20 Max Value 100 And Dy 18 1`] = `null`;

exports[`Storyshots Default.Single.Chart Two Groups Of Two Columns 1`] = `null`;

exports[`Storyshots Default.Single.Image Editor Of Image Content 1`] = `null`;

exports[`Storyshots Default.Single.Image Image Content 1`] = `null`;

exports[`Storyshots Default.Single.RichText Editor Of Multiline Content 1`] = `null`;

exports[`Storyshots Default.Single.RichText Editor Of Single Line Content 1`] = `null`;

exports[`Storyshots Default.Single.RichText Lists Content 1`] = `null`;

exports[`Storyshots Default.Single.RichText Multiline Content 1`] = `null`;

exports[`Storyshots Default.Single.RichText Single Line Content 1`] = `null`;

exports[`Storyshots Default.Single.SvgText HTML Text In SVG 1`] = `
<div>
  <p
    className="Typography-sc-1b4huth-0 cURStV"
    id="default-single-svgtext--html-text-in-svg"
    kind="Default.Single.SvgText"
    name="HTML Text In SVG"
    width="100%"
  >
    html document text
  </p>
  <div
    style={
      Object {
        "fontWeight": 600,
        "width": "100%",
      }
    }
  >
    should be rendered exactly as
  </div>
  <svg
    height="100%"
    width="100%"
  >
    <foreignObject
      className="SvgTypographyForeign-sc-10c5zbq-0 jImAGL"
      height="100%"
      width="100%"
      x={0}
      y={-1}
    >
      <p
        className="Typography-sc-1b4huth-0 SvgTypography__StyledTypography-jh05vp-0 SvgTypography__TextBox-jh05vp-1 lncloV"
        dominantBaseline="baseline"
        id="default-single-svgtext--html-text-in-svg"
        kind="Default.Single.SvgText"
        name="HTML Text In SVG"
        textAnchor="start"
        width={1}
      >
        html document text
      </p>
    </foreignObject>
  </svg>
</div>
`;

exports[`Storyshots Default.Single.SvgText SVG Text Width Property 1`] = `
<div>
  <p
    className="Typography-sc-1b4huth-0 fsrZYM"
    width={200}
  >
    text width is 200, extra is hidden
  </p>
  <div
    style={
      Object {
        "fontWeight": 600,
        "width": "100%",
      }
    }
  >
    should be rendered exactly as
  </div>
  <svg
    height="100%"
    width="100%"
  >
    <foreignObject
      className="SvgTypographyForeign-sc-10c5zbq-0 jImAGL"
      height="100%"
      width="100%"
      x={0}
      y={-1}
    >
      <p
        className="Typography-sc-1b4huth-0 SvgTypography__StyledTypography-jh05vp-0 SvgTypography__TextBox-jh05vp-1 lncloV"
        dominantBaseline="baseline"
        textAnchor="start"
        width={1}
      >
        text width is 200, extra is hidden
      </p>
    </foreignObject>
  </svg>
</div>
`;

exports[`Storyshots Default.Single.Table Editor Of Table 1 X 2 1`] = `null`;

exports[`Storyshots Default.Single.Table Table 1 X 2 1`] = `null`;

exports[`Storyshots Default.Single.Table Table 3 X 3 1`] = `null`;
